services/web-app/utils/schema.js

```diff
@@ -126,3 +127,40 @@ export const getLibraryVersionAsset = (str = '') => {
     asset
   }
 }
+
+/**
+ * Gets an array of tag names given an asset
+ * @param {import('../typedefs').Asset} asset
+ * @returns {string[]} An array of tag names
+ */
+export const getTagsList = (asset) => {
+  const tagNames = []
+
+  const tags = asset?.content?.tags ?? []
+
+  // Iterate over all tags
+  tags.forEach((tag) => {
+    // Look for a matching tag by asset type
+    const foundTag = get(tagsForType, `${asset?.content.type}.${tag}`)
+
+    if (foundTag) {
+      // It's a match, save it
+      tagNames.push(foundTag.name)
+    } else {
+      // It's not a match, see if it's part of a collection
+      Object.keys(tagsForCollection).forEach((collectionKey) => {
+        if (tags.includes(collectionKey)) {
+          // The asset is part of a collection, search in that collection's tags
+          const foundCollectionTag = get(tagsForCollection, `${collectionKey}.${tag}`)
+
+          if (foundCollectionTag) {
+            // It's a match, save it
+            tagNames.push(foundCollectionTag.name)
+          }
+        }
+      })
+    }
+  })
+
+  return tagNames
```

Comment:
```suggestion
  const tags = asset?.content?.tags ?? []

// retrieve valid tags for asset type
  const componentTypeTags = get(tagsForType, asset?.content.type, {})

// flatten all collections to obtain single object with all tags keys
  const allCollectionTags = Object.keys(tagsForCollection).reduce((prev, currKey) => {
    return Object.assign(prev, tagsForCollection[currKey])
  }, {})

// object containing all possible valid tags the asset can have given it's type
  const allPossibleTags = Object.assign(componentTypeTags, allCollectionTags)

// return array of tag names
  return tags.map(tag => allPossibleTags[tag]?.name).filter(val => !!val)
```
This saves us the potential nested iteration, just a suggestion

Reply:
I like it! I was considering doing something similar. What would happen if we happened to have a tag with the same key in componentTypeTags and allCollectionTags? Does the Object.assign pick a winner?

Reply:
yeah it'd pick the one that got added last into the object, so the one that is in allCollectionTags 
<img width="251" alt="image" src="https://user-images.githubusercontent.com/40550942/158631463-5eb4f9ca-1be9-4f2a-9d9d-2780a8482667.png">
Do we expect this to happen?

Reply:
Makes sense, even if there is an accidental duplicate, we would only want it to display once anyway.